src/main/java/io/confluent/kafkarest/AdminClientWrapper.java

```diff
@@ -0,0 +1,197 @@
+import io.confluent.kafkarest.entities.Topic;
+import io.confluent.rest.exceptions.RestServerErrorException;
+import jersey.repackaged.com.google.common.collect.ImmutableList;
+
+
+public class AdminClientWrapper {
+
+  private AdminClient adminClient;
+  private int initTimeOut;
+
+  public AdminClientWrapper(KafkaRestConfig kafkaRestConfig) {
+    Properties properties = new Properties(kafkaRestConfig.getAdminProperties());
+    properties.put(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaRestConfig.bootstrapBrokers());
+    adminClient = AdminClient.create(properties);
+    this.initTimeOut = kafkaRestConfig.getInt(KafkaRestConfig.KAFKACLIENT_INIT_TIMEOUT_CONFIG);
+  }
+
+  public List<Integer> getBrokerIds() {
+    List<Integer> brokerIds = new Vector<>();
+    DescribeClusterResult clusterResults = adminClient.describeCluster();
+    try {
+      Collection<Node>
+          nodeCollection =
+          clusterResults.nodes().get(initTimeOut, TimeUnit.MILLISECONDS);
+      for (Node node : nodeCollection) {
+        brokerIds.add(node.id());
+      }
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RestServerErrorException(Errors.KAFKA_ERROR_MESSAGE, Errors
+          .KAFKA_ERROR_ERROR_CODE, e);
+    }
+    return brokerIds;
+  }
+
+  public Collection<String> getTopicNames() {
+    Collection<String> allTopics = null;
+    try {
+      allTopics =
+          new TreeSet<>(adminClient.listTopics().names().get(initTimeOut, TimeUnit.MILLISECONDS));
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RestServerErrorException(Errors.KAFKA_ERROR_MESSAGE, Errors
+          .KAFKA_ERROR_ERROR_CODE, e);
+    }
+    return allTopics;
+  }
+
+  public boolean topicExists(String topic) {
+    try {
+      Collection<String> allTopics =
+          adminClient.listTopics().names().get(initTimeOut, TimeUnit.MILLISECONDS);
+      return allTopics.contains(topic);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RestServerErrorException(Errors.KAFKA_ERROR_MESSAGE, Errors
+          .KAFKA_ERROR_ERROR_CODE, e);
+    }
+  }
+
+  public Topic getTopic(String topicName) {
+    Topic topic = null;
+    try {
+      if (topicExists(topicName)) {
+        TopicDescription topicDescription = adminClient.describeTopics(
+            ImmutableList.<String>of(topicName)).values().get(topicName)
+            .get(initTimeOut, TimeUnit.MILLISECONDS);
+
+        topic = buildTopic(topicName, topicDescription);
+      }
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RestServerErrorException(Errors.KAFKA_ERROR_MESSAGE, Errors
+          .KAFKA_ERROR_ERROR_CODE, e);
+    }
+    return topic;
+  }
+
+  public List<Partition> getTopicPartitions(String topicName) {
+    List<Partition> partitions = null;
+    try {
+      TopicDescription topicDescription = adminClient.describeTopics(
```

Comment:
describeTopics and associated exception handling can probably be factored out for terser code.